simple-generated/src/event.rs

```diff
@@ -1,87 +1,116 @@
-use crate::app::AppResult;
-use ratatui::crossterm::event::{
-    self, Event as CrosstermEvent, KeyEvent, KeyEventKind, MouseEvent,
-};
+use crossterm::event::{self, Event as CrosstermEvent, KeyEvent, KeyEventKind, MouseEvent};
 use std::sync::mpsc;
 use std::thread;
 use std::time::{Duration, Instant};
 
 /// Terminal events.
-#[derive(Clone, Copy, Debug)]
+#[derive(Debug)]
 pub enum Event {
-    /// Terminal tick.
+    /// The tick event is sent at a regular interval and can be used to trigger application state
+    /// updates or animations.
     Tick,
+
     /// Key press.
     Key(KeyEvent),
+
     /// Mouse click/scroll.
     Mouse(MouseEvent),
+
     /// Terminal resize.
+    ///
+    /// This event may be fired multiple times in quick succession, so it is recommended not to
+    /// render the UI on every resize event.
     Resize(u16, u16),
+
+    /// An error occurred.
+    Error(std::io::Error),
 }
 
 /// Terminal event handler.
-#[allow(dead_code)]
+///
+/// This struct is responsible for listening to terminal events and sending them to the main thread.
+///
+/// It sends a tick event at a regular interval. The tick rate is specified by the `tick_rate`
+/// parameter. If an error occurs, it sends an error event to the main thread and then stops
+/// running.
 #[derive(Debug)]
-pub struct EventHandler {
-    /// Event sender channel.
-    sender: mpsc::Sender<Event>,
+pub struct EventSource {
```

Comment:
Maybe `EventListener`? 

I actually like `EventHandler`, because it technically does handle getting the events and allows the user to retrieve them after it gets them.

I asked chatgpt for various names:

<img width="850" alt="image" src="https://github.com/user-attachments/assets/d9f2d17a-7333-4e8d-bf46-cf81c71da0df">

Reply:
Removed the entire module in 6a76940